### PR TITLE
Add retry logic to support temporary redis cluster failure (\Predis\C…

### PR DESCRIPTION
**Pull request by digital2real**
This changes are trying to complete the retry logic in order to increase the tolerance to a temporary unavailability of a redis cluster.

(Sorry for previous PR #731 on main)

### CODE REVIEW

**Comment by tillkruss** [comment]
No worries.

@nrk Any input on this?

**Comment by digital2real** [comment]
Hi,
@tillkruss , @nrk do you have any comment/update on this?

File: src/Connection/Aggregate/RedisCluster.php [hunk 1]
```diff
@@ -233,6 +235,7 @@ private function queryClusterNodeForSlotsMap(NodeConnectionInterface $connection
                 }
 
                 ++$retries;
+		        sleep(2^$retries);
```
Comment (tillkruss) [inline src/Connection/Aggregate/RedisCluster.php:238]: Indentation seems off.

File: src/Connection/Aggregate/RedisCluster.php [hunk 2]
```diff
@@ -545,22 +548,37 @@ private function retryCommandOnFailure(CommandInterface $command, $method)
     {
         $failure = false;
 
+        $retries= 0 ;
```
Comment (tillkruss) [inline src/Connection/Aggregate/RedisCluster.php:551]: Needs formatting.

File: src/Connection/Aggregate/RedisCluster.php [hunk 3]
```diff
@@ -545,22 +548,37 @@ private function retryCommandOnFailure(CommandInterface $command, $method)
...
+
         RETRY_COMMAND: {
             try {
                 $response = $this->getConnection($command)->$method($command);
-            } catch (ConnectionException $exception) {
-                $connection = $exception->getConnection();
-                $connection->disconnect();
-
-                $this->remove($connection);
+                if (is_a($response, '\\Predis\\Response\\Error')){
```
Comment (tillkruss) [inline src/Connection/Aggregate/RedisCluster.php:556]: Use `instanceof` please.

File: src/Connection/Aggregate/RedisCluster.php [hunk 4]
```diff
@@ -545,22 +548,37 @@ private function retryCommandOnFailure(CommandInterface $command, $method)
...
-
-                $this->remove($connection);
+                if (is_a($response, '\\Predis\\Response\\Error')){ 
+                    $message = $response->getMessage() ;
+                    if (strpos($message, "CLUSTERDOWN") === 0 ) {
+                        sleep (2^($retries+1));
+                        throw new ServerException($message) ;  
+                    }
+                }
+            } catch (ConnectionException|ServerException $exception) {
```
Comment (tillkruss) [inline src/Connection/Aggregate/RedisCluster.php:563]: I don't think this syntax is compatible with olde PHP versions.

**Review by tillkruss — CHANGES_REQUESTED** [review]

**Comment by tillkruss** [comment]
Feel free to re-submit to the 2.0 `main` branch.

**Pull request closed without merging**